### PR TITLE
TFT: Tilt has a button that allows user to create a snapshot and share the link

### PR DESCRIPTION
**Pull request by mariavivanco**
Screen Shot 2019-08-13 at 5 40 52 PM
Screen Shot 2019-08-13 at 5 41 01 PM

finally works end to end ! 😎

### CODE REVIEW

File: internal/hud/server/server.go [hunk 1]
```diff
@@ -266,3 +271,81 @@ func tsAlertToBackendAlert(alert tsAlert) tft.Alert {
...
+	return fmt.Sprintf("https://%s/snapshot/%s", tiltAlertsDomain, id)
+}
+
+func (s *HeadsUpServer) HandleNewSnapshot(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		http.Error(w, "must be POST request", http.StatusBadRequest)
+		return
+	}
+
+	request, err := http.NewRequest(http.MethodPost, "https://alerts.tilt.dev/api/snapshot/new", req.Body) // New request
```
Comment (jazzdan) [inline internal/hud/server/server.go]:
```suggestion
	request, err := http.NewRequest(http.MethodPost, "https://alerts.tilt.dev/api/snapshot/new", req.Body)
```

File: internal/hud/server/server_test.go [hunk 2]
```diff
@@ -21,6 +24,16 @@ import (
...
 )
 
+var originalWD string
+
+func init() {
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	originalWD = wd
```
Comment (jazzdan) [inline internal/hud/server/server_test.go]: Hmm why do we need to do this? Does the HUD server test muck with the current working directory?

File: internal/tft/client/client.go [hunk 3]
```diff
@@ -54,6 +54,8 @@ func (t *tftClient) SendAlert(ctx context.Context, alert Alert) (AlertID, error)
 	return AlertID(r.ID), nil
 }
 
+//TODO TFT: Create send snapshot function: returns snapshot ID and error
```
Comment (jazzdan) [inline internal/tft/client/client.go]: Are we still planning on doing this?

File: web/src/HUD.tsx [hunk 4]
```diff
@@ -167,6 +182,25 @@ class HUD extends Component<HudProps, HudState> {
       .then(err => console.error(err))
   }
 
+  sendSnapshot(snapshot: Snapshot) {
+    let url = `//${window.location.host}/api/snapshot/new`
+    fetch(url, {
+      method: "post",
+      body: JSON.stringify(snapshot),
+    })
+      .then(res => {
```
Comment (jazzdan) [inline web/src/HUD.tsx]: I think this promise could be re-written more idiomatically as:

```typescript
fetch()
  .then(res => res.json)
  .then((value: NewSnapshotResponse) => {
            this.setState({
              SnapshotLink: value.url,
            })
  })
  .catch(err => console.error(err)
```

then the same catch will handle everything.

File: web/src/TopBar.tsx [hunk 5]
```diff
@@ -35,4 +48,32 @@ class TopBar extends PureComponent<TopBarProps> {
...
+        <button
+          title="Open link in new tab"
+          onClick={() => window.open(snapshotURL)}
+        >
+          Open
+        </button>
+      </section>
+    )
+  }
+  //TODO TFT - formatting the button
```
Comment (jazzdan) [inline web/src/TopBar.tsx]: Hmm the button looks formatted to me!

File: internal/hud/server/server_test.go (hunk 2 above)
Comment (hyu) [reply internal/hud/server/server_test.go]: Haha, as far as I know we were just trying to read the testdata dump with all the JSON, and found this snippet in some SO answer.

**Pull request merged**